hypershift-operator/controllers/scheduler/azure/controllers_test.go

```diff
@@ -0,0 +1,212 @@
+package azure
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	. "github.com/onsi/gomega"
+
+	hyperv1 "github.com/openshift/hypershift/api/hypershift/v1beta1"
+	schedulingv1alpha1 "github.com/openshift/hypershift/api/scheduling/v1alpha1"
+	hyperapi "github.com/openshift/hypershift/support/api"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func TestAzureSchedulerReconcile(t *testing.T) {
```

Comment:
nit: this is within the scheduler/azure, so TestReconcile seems should be sufficient